javascript/matrix: suggest to defer value calculation to it's getter (#2053)
exercism/website-copy: tracks/javascript/exercises/matrix/mentoring.md

    )
  })
```

>  -If a student try to do everything inside the constructor suggest on them to defer calculations to getters as they are supposed to defer the cost of calculating the value until it is needed. If the value never needed, you never pay the cost.[[2](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get)] 

Reviewer: I would suggest a few minor changes:

```suggestion
- If a student tries to implement everything inside a constructor, suggest on them to defer calculations to getters as they are supposed to defer the cost of calculating the result until it is needed. [[2](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get)] 
```

Author: Thanks

> - If a student tries to implement everything inside a constructor, suggest on them to defer calculations to getters as they are supposed to defer the the cost of calculating the value until the value is needed. If it is never needed, you never pay the cost. [[2](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get)] 

Reviewer 2: Consistency issue with line 41-52, see https://github.com/exercism/website-copy/pull/2053#issuecomment-917553347

Author: I agree that those lines  are inconsistent. They are inconsistent with the **exercise's start state** which clearly gives the user getter skeletons to solve the exercise with! While in the mentor notes it proposes a different solution and THEN (lines 50-52) says the user MAY use getters to solve the problem..

Reviewer 2: @salouri I wasn't aware the start state (stub file) was changed like that. (I am sorry, I was absent from Exercism for a while.) You are totally right, the mentor notes should be consistent with the stub. So if you could make this PR a somewhat bigger update so that the notes match the stub again, that would be great!
